src/core/Akka.Streams/Dsl/Graph.cs

```diff
@@ -424,6 +424,188 @@ public MergePreferred(int secondaryPorts, bool eagerClose = false)
+                        else
+                        {
+                            runningUpstreams -= 1;
+                            if (UpstreamsClosed && !HasPending) CompleteStage();
+                        }
+                    });
+                }
+
+                SetHandler(_stage.Out, this);
+            }
+
+            public override void PreStart() => _stage.In.ForEach(TryPull);
+
+            public override void OnPull()
+            {
+                if (HasPending)
+                    DequeueAndDispatch();
+            }
+
+            public bool HasPending => allBuffers.Any(c => c.NonEmpty);
+
+            public bool UpstreamsClosed => runningUpstreams == 0;
+
+            private void DequeueAndDispatch()
+            {
+                var input = SelectNextElement();
+                Push(_stage.Out, Grab(input));
+                if (UpstreamsClosed && !HasPending)
+                    CompleteStage();
+                else
+                    TryPull(input);
+            }
+
+            private Inlet<T> SelectNextElement()
+            {
+                var tp = 0;
+                var ix = 0;
+
+                while (ix < _stage.In.Count)
+                {
+                    if (allBuffers[ix].NonEmpty)
+                    {
+                        tp += _stage.Priorities[ix];
+                    }
+                    ix += 1;
+                }
+
+                int r = randomGen.Next(tp);
+                Inlet<T> next = null;
+                ix = 0;
+
+                while (ix < _stage.In.Count && next == null)
+                {
+                    if (allBuffers[ix].NonEmpty)
+                    {
+                        r -= _stage.Priorities[ix];
+                        if (r < 0)
+                            next = allBuffers[ix].Dequeue();
+                    }
+                    ix += 1;
+                }
+
+                return next;
+            }
+
+            public override string ToString() => "MergePrioritized";
+        }
+        #endregion
+
+        /// <summary>
+        /// Create a new <see cref="T:Akka.Streams.Dsl.MergePrioritized`1" /> with specified number of input ports.
+        /// </summary>
+        /// <param name="priorities">Priorities of the input ports</param>
+        public MergePrioritized(IList<int> priorities)
+            :this(priorities, false)
+        {
+            
+        }
+
+        /// <summary>
+        /// Create a new <see cref="T:Akka.Streams.Dsl.MergePrioritized`1" /> with specified number of input ports.
+        /// </summary>
+        /// <param name="priorities">Priorities of the input ports</param>
+        /// <param name="eagerComplete">If true, the merge will complete as soon as one of its inputs completes</param>
+        /// <exception cref="ArgumentException">
+        /// This exception is thrown when the specified <paramref name="priorities"/> is less or equal zero.
+        /// </exception>
```

Comment:
Exception should be also visible on the other constructor or maybe set eagerComplete to false as default and use single constructor